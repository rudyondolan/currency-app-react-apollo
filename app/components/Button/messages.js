/*
 * Button Messages
 *
 * This contains all the text for the Button component.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
  add: {
    id: 'app.components.Button.add',
    defaultMessage: 'Add More Currencies',
  },
});
