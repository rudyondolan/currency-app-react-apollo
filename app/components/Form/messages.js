/*
 * Form Messages
 *
 * This contains all the text for the Form component.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
  submit: {
    id: 'app.components.Form.submit',
    defaultMessage: 'Submit',
  },
});
