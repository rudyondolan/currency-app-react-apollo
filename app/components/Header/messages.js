/*
 * Header Messages
 *
 * This contains all the text for the Header component.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
  list: {
    id: 'boilerplate.components.Header.list',
    defaultMessage: 'List Currency',
  },
  about: {
    id: 'boilerplate.components.Header.about',
    defaultMessage: 'About',
  },
});
