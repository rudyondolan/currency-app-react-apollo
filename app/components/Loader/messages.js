/*
 * Loader Messages
 *
 * This contains all the text for the Loader component.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
  loading: {
    id: 'app.components.Loader.loading',
    defaultMessage: 'Loading...',
  },
});
