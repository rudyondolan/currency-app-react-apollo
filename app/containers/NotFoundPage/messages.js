/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  page: {
    id: 'boilerplate.containers.NotFoundPage.page',
    defaultMessage: 'Page not found.',
  },
});
