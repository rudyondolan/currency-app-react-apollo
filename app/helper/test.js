import React from 'react';
import { IntlProvider, intlShape } from 'react-intl';
import { mount, shallow } from 'enzyme';

const intlProvider = new IntlProvider({ locale: 'en' }, {});
const { intl } = intlProvider.getChildContext();

function nodeWithIntlProp(node) {
  return React.cloneElement(node, { intl });
}

export function shallowWithIntl(node, { context, ...additionalOptions } = {}) {
  return shallow(nodeWithIntlProp(node), {
    context: Object.assign({}, context, { intl }),
    ...additionalOptions,
  });
}

export function mountWithIntl(
  node,
  { context, childContextTypes, ...additionalOptions } = {},
) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, { intl }),
    childContextTypes: Object.assign(
      {},
      { intl: intlShape },
      childContextTypes,
    ),
    ...additionalOptions,
  });
}
