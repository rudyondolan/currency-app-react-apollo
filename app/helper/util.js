export function toFloat(value) {
  if (typeof value === 'string') {
    const fVal = parseFloat(value);
    if (typeof fVal === 'number' && !isNaN(fVal)) {
      return fVal;
    }

    return 1;
  }

  return value;
}

