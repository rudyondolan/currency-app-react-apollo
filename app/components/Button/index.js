import React from 'react';
import { Button } from 'semantic-ui-react';
import PropTypes from 'prop-types';

import { FormattedMessage } from 'react-intl';
import messages from './messages';

function ExchangeButton({ onClick, ...rest }) {
  return (
    <Button {...rest} onClick={() => onClick()}>
      <FormattedMessage {...messages.add} />
    </Button>
  );
}

ExchangeButton.propTypes = {
  color: PropTypes.string,
  size: PropTypes.string,
  onClick: PropTypes.func,
};

export default ExchangeButton;
