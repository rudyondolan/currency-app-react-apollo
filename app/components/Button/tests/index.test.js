import React from 'react';
import { Button as ButtonSemantic } from 'semantic-ui-react';
import { mountWithIntl } from 'helper/test';
import Button from '../index';
import messages from '../messages';

describe('<Button />', () => {
  test('should match the snapshot', () => {
    const wrapper = renderButton();
    expect(wrapper).toMatchSnapshot();
  });

  test('should match text button with the messages', () => {
    const wrapper = renderButton();
    expect(wrapper.find(ButtonSemantic).text()).toBe(
      messages.add.defaultMessage,
    );
  });

  test('should call click event', () => {
    const onClick = jest.fn();
    const props = { onClick };
    const wrapper = renderButton(props);
    wrapper.find(ButtonSemantic).simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});

function renderButton(props = {}) {
  const propsToUse = {
    size: 'large',
    color: 'teal',
    onClick() {},
    ...props,
  };

  return mountWithIntl(<Button {...propsToUse} />);
}
