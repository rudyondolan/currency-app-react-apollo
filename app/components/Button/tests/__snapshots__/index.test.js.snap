// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match the snapshot 1`] = `
<ExchangeButton
  color="teal"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
  onClick={[Function]}
  size="large"
>
  <Button
    as="button"
    color="teal"
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {},
        "now": [Function],
        "textComponent": "span",
      }
    }
    onClick={[Function]}
    role="button"
    size="large"
  >
    <button
      className="ui teal large button"
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": Object {},
          "now": [Function],
          "textComponent": "span",
        }
      }
      onClick={[Function]}
      role="button"
    >
      <FormattedMessage
        defaultMessage="Add More Currencies"
        id="app.components.Button.add"
        values={Object {}}
      >
        <span>
          Add More Currencies
        </span>
      </FormattedMessage>
    </button>
  </Button>
</ExchangeButton>
`;
